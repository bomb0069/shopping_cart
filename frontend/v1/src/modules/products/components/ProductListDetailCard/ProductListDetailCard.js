import React from 'react'
import Grid from '@material-ui/core/Grid'

import ProductDetailCard from '../ProductDetailCard'

const ProductListDetailCard = ({
  productItems
}) => {
  return (
    <Grid
      container
      spacing={5}
      style={{marginTop: '2em'}}
    >
      {productItems.map(item => (
        <Grid
          item
          key={item.id}
          sm={4}
          xs={12}
        >
          <ProductDetailCard
                
            {...item}
          />
        </Grid>
      ))}
    </Grid>
  )
}

export default ProductListDetailCard
