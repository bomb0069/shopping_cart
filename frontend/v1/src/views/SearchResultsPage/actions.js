import axios from 'axios'

export const getProductWithFilter = async (config) => {
  const response = await axios.get('http://localhost:3000/api/v1/products?age=3_to_5', config)
              
  return response
}