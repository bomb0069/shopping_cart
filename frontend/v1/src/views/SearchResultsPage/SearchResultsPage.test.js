import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import SearchResultsPage from './SearchResultsPage'
import Header from '../../modules/common/Header'
import ProductListDetailCard from '../../modules/products/components/ProductListDetailCard'


describe('<SearchResultsPage />', () => {
  let component

  beforeEach(() => {
    component = shallow(<SearchResultsPage />)

  })

  it('should render with no errors', () => {
    
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render with Header', () => {
    expect(toJson(component)).toMatchSnapshot()

    expect(component.find(Header).length).toEqual(1)
  })

  it('should render with ProductListDetailCard', () => {
    expect(toJson(component)).toMatchSnapshot()

    expect(component.find(ProductListDetailCard).length).toEqual(1)
  })
})