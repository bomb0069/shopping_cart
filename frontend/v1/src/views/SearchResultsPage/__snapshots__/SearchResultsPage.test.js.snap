// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResultsPage /> should render with Header 1`] = `
<div>
  <WithStyles(Header)
    amount={0}
    handleSubmit={[Function]}
  />
  <ProductListDetailCard
    productItems={Array []}
  />
</div>
`;

exports[`<SearchResultsPage /> should render with ProductListDetailCard 1`] = `
<div>
  <WithStyles(Header)
    amount={0}
    handleSubmit={[Function]}
  />
  <ProductListDetailCard
    productItems={Array []}
  />
</div>
`;

exports[`<SearchResultsPage /> should render with no errors 1`] = `
<div>
  <WithStyles(Header)
    amount={0}
    handleSubmit={[Function]}
  />
  <ProductListDetailCard
    productItems={Array []}
  />
</div>
`;
